packages/devtools-components/src/tree.js

```diff
@@ -554,22 +532,23 @@ const Tree = createClass({
    *        The item to be focused, or undefined to focus no item.
    */
   _focus(index, item) {
-    if (item !== undefined) {
-      const itemStartPosition = index * this.props.itemHeight;
-      const itemEndPosition = (index + 1) * this.props.itemHeight;
-
-      // Note that if the height of the viewport (this.state.height) is less than
-      // `this.props.itemHeight`, we could accidentally try and scroll both up and
-      // down in a futile attempt to make both the item's start and end positions
-      // visible. Instead, give priority to the start of the item by checking its
-      // position first, and then using an "else if", rather than a separate "if",
-      // for the end position.
-      if (this.state.scroll > itemStartPosition) {
-        this.refs.tree.scrollTop = itemStartPosition;
-      } else if ((this.state.scroll + this.state.height) < itemEndPosition) {
-        this.refs.tree.scrollTop = itemEndPosition - this.state.height;
-      }
-    }
+    // TODO: Revisit how we should handle focus without having fixed item height.
```

Comment:
hmm, so _focus is important in two cases: 

1. revealing a node and scrolling to it
2. when the user arrows down repeatedly and keeping the focus in view

will these two use cases be supported?

Reply:
With this PR, in this state, no.
Do you think we should handle this here or as part of https://github.com/devtools-html/devtools-core/issues/690 ?
This PR fixes a pretty important perf regression on the console, and I'd like to get it pretty soon.

Reply:
Lets land this and then fix #690 before the debugger pulls in the patch.